pkg/mutate/squashfs.go

```diff
@@ -104,23 +123,57 @@ func SquashfsLayer(base v1.Layer, dir string, opts ...SquashfsConverterOpt) (v1.
 }
 
 // makeSquashfs returns the path to a squashfs file that contains the contents of the uncompressed
-// TAR stream from r.
-func (c *squashfsConverter) makeSquashfs(r io.Reader) (string, error) {
+// TAR stream from l.Uncompressed().
+func (c *squashfsConverter) makeSquashfs(l v1.Layer) (string, error) {
```

Comment:
Calling `l.Uncompressed` twice will work in certain situations, but it can be expensive, for example if `l` is coming from a `remote` it would end up using twice the bandwidth.

One way to workaround this would be to use a temporary file in `c.dir` and a `io.TeeReader` to write the TAR and complete your first scan with `scanAUFSOpaque`. Then in your second pass you'd process the file with `whiteoutFilter`?

I'm also wondering if it might be simpler to keep `makeSquashfs` focussed on the TAR->SquashFS conversion, and have `SquashfsLayer` handle pre-processing the input TAR stream if `c.convertWhiteout` is set. I could envision a `func (*squashfsConverter).convertWhiteout(io.Reader) (io.Reader, error)` that is focussed on the whiteout conversion.